Profile picture added to Feature/admin users

KianMiridoozini opened the pull request:
User management rows now include the user's respective profile picture that links to the user's profile page

KianMiridoozini committed:
Profile picture is now included in the user-management rows.

Profile pictures link to the users profile page.

KianMiridoozini committed:
Merge branch 'develop' of https://github.com/jayb2302/Drinx into feature/admin-users

jayb2302 commented on app/repositories/UserRepository.php:
```diff
@@ -228,24 +228,39 @@ public function searchUsers($query) {
-            WHERE u.username LIKE :query
-        ");
-        $stmt->execute(['query' => '%' . $query . '%']);
+            LEFT JOIN user_profile p ON u.user_id = p.user_id
```
This function looks longer than it needs to be, is there a reason for that ?

KianMiridoozini replied:
We wanted to implement javascript for some of the related functionality, this means some extra lifting from the php end, in this case the query used in the repository for this function.

This function is designed to fetch user data with optional filtering, handling account status labels and profile pictures in one query:

SQL Query: Selects user details, with:

CASE for readable account status ("Active," "Suspended," etc.).
COALESCE for default profile picture if missing.
LEFT JOIN to include profile picture data without excluding users without one.
Query Filtering: If a query (username search) is provided, a WHERE clause filters results to match the search term.

jayb2302 commented on app/views/admin/manage_users.php:
```diff
@@ -15,7 +15,9 @@
-                    <a href="/profile/<?= htmlspecialchars($user->getUsername()); ?>" class="view-profile">View Profile</a>
+                    <a href="/profile/<?= htmlspecialchars($user->getUsername()); ?>" class="view-profile">
+                        <img class="" alt="">
+                    </a>
```
no class and no alt

KianMiridoozini replied:
The class an alt are implemented through javascript.

As seen upon inspecting the page, profile pictures do have class and alt text.

jayb2302 commented on public/assets/js/admin.js:
```diff
@@ -10,14 +10,25 @@ document.addEventListener("DOMContentLoaded", () => {
+                             width="40" 
+                             height="40">
+                    </a>
+                </td>
```
I see its here :P 
but we also have a class for user pictures, we should try to reuse classes

KianMiridoozini replied:
I fail to find a class called user picture. The closest instance i found is  .control-user-picture

KianMiridoozini committed:
feature/admin-users

Adjustments have been made to:
- Query handling for searchAllUsers.
- Class and Alt text handling; from admin.js to manage-users.php
